File: 2396-strictly-palindromic-number/2396-strictly-palindromic-number.py
class Solution(object):
    def checkPalindrome(self, s):
        i, j = 0, len(s) - 1
        while i<=j :
            if s[i] != s[j]:
                return False
            i += 1
            j -= 1
        return True
    
    def toBase(self, n, base):
        res = ""
        while n > 0:
            res = str(n % base) + res
            n //= base
        return self.checkPalindrome(res)
        
    def isStrictlyPalindromic(self, n):
        for i in range(2, n-1):
            if not self.toBase(n, i):
                return False
        return True
        

File: 2396-strictly-palindromic-number/2396-strictly-palindromic-number.cpp
class Solution {
public:
    bool checkPalindrome(string n){
        for(int i= 0, j = n.size()-1; i<=j; i++, j--){
            if(n[i] != n[j])
                return false;
        }
        return true;
    }

    bool toBase(int n, int base){
        string res = "";
        while(n>0){
            res = to_string(n%base) + res;
            n /= base;
             cout << res << endl;
        }
        return checkPalindrome(res);
    }
    bool isStrictlyPalindromic(int n) {
        for(int i = 2; i <= n-2; i++){
            if(!toBase(n, i))return false;
        }
        return true;
    }
};